assets/kube-state-metrics/custom-resource-state-configmap.yaml

```diff
@@ -0,0 +1,561 @@
+apiVersion: v1
+data:
+  custom-resource-state-configmap.yaml: |-
+    "kind": "CustomResourceStateMetrics"
+    "spec":
+      "resources":
+      - "groupVersionKind":
+          "group": "autoscaling.k8s.io"
+          "kind": "VerticalPodAutoscaler"
+          "version": "v1"
+        "metrics":
+        - "commonLabels": null
+          "each":
+            "info":
+              "labelsFromPath":
+                "*":
+                - "metadata"
+                - "annotations"
+                "container":
+                - "containerName"
+              "valueFrom": 1
+            "type": "Info"
+          "help": "Kubernetes annotations converted to Prometheus labels."
+          "labelsFromPath":
+            "namespace":
+            - "metadata"
+            - "namespace"
+            "target_api_version":
+            - "spec"
+            - "targetRef"
+            - "apiVersion"
+            "target_kind":
+            - "spec"
+            - "targetRef"
+            - "kind"
+            "target_name":
+            - "spec"
+            - "targetRef"
+            - "name"
+            "verticalpodautoscaler":
+            - "metadata"
+            - "name"
+          "name": "kube_verticalpodautoscaler_annotations_info"
+        - "commonLabels": null
+          "each":
+            "info":
+              "labelsFromPath":
+                "*":
+                - "metadata"
+                - "labels"
+                "container":
+                - "containerName"
+              "valueFrom": 1
+            "type": "Info"
+          "help": "Kubernetes labels converted to Prometheus labels."
+          "labelsFromPath":
+            "namespace":
+            - "metadata"
+            - "namespace"
+            "target_api_version":
+            - "spec"
+            - "targetRef"
+            - "apiVersion"
+            "target_kind":
+            - "spec"
+            - "targetRef"
+            - "kind"
+            "target_name":
+            - "spec"
+            - "targetRef"
+            - "name"
+            "verticalpodautoscaler":
+            - "metadata"
+            - "name"
+          "name": "kube_verticalpodautoscaler_labels_info"
```

Comment:
ACK, the `*` wildcard should expose all labels (for eg., https://github.com/kubernetes/kube-state-metrics/blob/main/docs/customresourcestate-metrics.md#multiple-metricskitchen-sink).

Reply:
For other `kube_*_labels_info` metrics, we don't expose all Kubernetes labels because of cardinality concerns (minus a few exceptions like `for pods, namespaces, ...). I'd probably go the same way for VPA metrics until we have a request/use case to do differently.

@jan--f WDYT?

Reply:
ACK, I'll reopen and update https://issues.redhat.com/browse/MON-3191 with the discussion here.

Reply:
We may want to revisit the regex here (to define behaviors for all `kube_customresource_.+_annotation_info` metrics, and disable annotations (and labels) entirely: https://github.com/openshift/cluster-monitoring-operator/blob/8db648fa23ade5453b87a1f58206f07be5791110/assets/kube-state-metrics/deployment.yaml#L41

Joel's mail mentioned a particular ask for `verticalpodautoscaler_status_recommendation_containerrecommendations_target_cpu`, so leaving out the annotation and labels metrics should be safe from that POV. 
```This did exactly what I wanted:
              - name: "verticalpodautoscaler_status_recommendation_containerrecommendations_target_cpu"
                each:
                  type: Gauge
                  gauge:
                    path: [status, recommendation, containerRecommendations]
                    valueFrom: [target, cpu]
                    labelsFromPath:
                      container: [containerName]
                  commonLabels:
                    resource: "cpu"
                    unit: "core"
``` 
Although, for the future, I would want `metric-labels-allowlist` to support custom resources as well. Alternatively, we could add options to `KubeStateMetricsConfig` to allow users to explicitly enable all VPA labels and annotations metrics, albeit that could steer them into a crash state, so hardcoding on our end makes sense. That is, once `metric-labels-allowlist` supports custom resources upstream, [we can add `verticalpodautoscalers=[<specific labels>]`](https://github.com/openshift/cluster-monitoring-operator/blob/8db648fa23ade5453b87a1f58206f07be5791110/assets/kube-state-metrics/deployment.yaml#L42C13-L42C36) (a set of nominal ones and others on a case-by-case request basis).